File: Python/day1.py
def main():
    print(part1())
    print(part2())


def part1():
    ans = 0
    input = open("../inputs/day1input.txt")
    directions = input.read()
    for c in directions:
        if c == "(":
            ans += 1
        elif c == ")":
            ans -= 1
    return ans


def part2():
    ans = 0
    input = open("../inputs/day1input.txt")
    directions = input.read()
    for i, c in enumerate(directions):
        if c == "(":
            ans += 1
        elif c == ")":
            ans -= 1
        if ans < 0:
            return i + 1


if __name__ == "__main__":
    main()

File: Java/Day1.java
package Java;

import java.io.FileNotFoundException;
import java.util.ArrayList;

class Day1 {
    public static void main(String[] args) {
        try {
            ArrayList<String> data = Helpers.readFile("./inputs/day1input.txt");
            String input = data.get(0);
            System.out.println(part1(input));
            System.out.println(part2(input));

        } catch (FileNotFoundException e) {
        }
    }

    public static int part1(String input) {
        int ans = 0;
        for (int i = 0; i < input.length(); i++) {
            char c = input.charAt(i);
            if (c == '(') {
                ans++;
            } else if (c == ')') {
                ans--;
            }
        }
        return ans;
    }

    public static int part2(String input) {
        int ans = 0;
        for (int i = 0; i < input.length(); i++) {
            char c = input.charAt(i);
            if (c == '(') {
                ans++;
            } else if (c == ')') {
                ans--;
            }
            if (ans < 0) {
                return i + 1;
            }
        }
        return ans;
    }
}
